### PR TITLE
Issue 777 - refactored tests for Ehcache

### PR DESCRIPTION
**Pull request by AbfrmBlr**

### CODE REVIEW

**Comment by ljacomet** [comment]
The following tests still need adaptations / split:
* EhcacheBasicIteratorTest
* EhcacheBasicPutAllTest
* EhcacheBasicRemoveAllTest

File: core/src/test/java/org/ehcache/core/EhcacheWithLoaderWriterBasicClearTest.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.core;
+
+import org.ehcache.Status;
+import org.ehcache.core.EhcacheWithLoaderWriter;
```
Comment (ljacomet) [inline core/src/test/java/org/ehcache/core/EhcacheWithLoaderWriterBasicClearTest.java]: Import from same package - remove

File: core/src/test/java/org/ehcache/core/EhcacheWithLoaderWriterBasicContainsKeyTest.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.core;
+
+import org.ehcache.Status;
+import org.ehcache.core.EhcacheWithLoaderWriter;
```
Comment (ljacomet) [inline core/src/test/java/org/ehcache/core/EhcacheWithLoaderWriterBasicContainsKeyTest.java]: EhcacheBasicRemoveAllTest

File: core/src/test/java/org/ehcache/core/EhcacheWithLoaderWriterBasicGetTest.java [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.core;
+
+import org.ehcache.Status;
+import org.ehcache.core.EhcacheWithLoaderWriter;
```
Comment (ljacomet) [inline core/src/test/java/org/ehcache/core/EhcacheWithLoaderWriterBasicGetTest.java]: Import from same package - remove

File: core/src/test/java/org/ehcache/core/EhcacheWithLoaderWriterBasicRemoveTest.java [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.core;
+
+import org.ehcache.Status;
+import org.ehcache.core.EhcacheWithLoaderWriter;
```
Comment (ljacomet) [inline core/src/test/java/org/ehcache/core/EhcacheWithLoaderWriterBasicRemoveTest.java]: Import from same package - remove

File: core/src/test/java/org/ehcache/core/EhcacheWithLoaderWriterBasicRemoveValueTest.java [hunk 5]
```diff
@@ -0,0 +1,849 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.core;
+
+import org.ehcache.Status;
+import org.ehcache.core.EhcacheWithLoaderWriter;
```
Comment (ljacomet) [inline core/src/test/java/org/ehcache/core/EhcacheWithLoaderWriterBasicRemoveValueTest.java]: Import from same package - remove

**Comment by AbfrmBlr** [comment]
Rebased with all tests.

**Comment by ljacomet** [comment]
`EhcacheBasicGetAllTest`, `EhcacheBasicPutAllTest`, `EhcacheBasicRemoveAllTest` and `EhcacheBasicIteratorTest` still import `CacheLoaderWriter` which seems wrong.

I saw that the `getEhcache(CacheLoaderWriter)` method is still there in some of these classes, just being passed `null` and failing if different than `null`. It would be cleaner to get rid of that parameter altogether so that there is no doubt.

**Pull request merged**